src/enable_variable_ops/ngraph_rewrite_for_tracking.cc

```diff
@@ -0,0 +1,298 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/graph/node_builder.h"
+#include "tensorflow/core/graph/types.h"
+
+#include "ngraph_rewrite_for_tracking.h"
+#include "ngraph_utils.h"
+
+using namespace std;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// TODO(Mingshsan): Rewrite the Replace* function to a helper function
+Status ReplaceNGraphVariable(Graph* graph, Node* node, Node** replacement,
+                             std::string node_new_name, bool just_looking,
+                             bool outputs_ng_supported, int graph_id) {
+  NGRAPH_VLOG(1) << "Replacing NGraphVariable " << node->name();
+
+  TensorShape shape;
+  DataType dtype;
+  TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "shape", &shape));
+  TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "dtype", &dtype));
+
+  std::string container;
+  std::string shared_name;
+  // int graph_id;
+  std::string backend_name;
+
+  if (GetNodeAttr(node->attrs(), "container", &container) != Status::OK()) {
+    container = "";
+  }
+  if (GetNodeAttr(node->attrs(), "shared_name", &shared_name) != Status::OK()) {
+    shared_name = "";
+  }
+
+  // TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "ngraph_graph_id",
+  // &graph_id));
+
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(node->attrs(), "_ngraph_backend", &backend_name));
+
+  // NGRAPHVARIABLE
+  TF_RETURN_IF_ERROR(
+      NodeBuilder(node_new_name, node->type_string())
```

Comment:
same comment about new name

Reply:
Suggestion: Can we create a utility function that takes in all that is needed to build a node like an Attr list, datatype etc does the node building since we are using this code a lot.

Reply:
Good suggestion. Currently the replacement is already in a utility function. There are two functions Replace NGraphVariable and NGraphAssign. In this PR, i will try to refactor to use these functions in CaptureVariablePass also, where we are replacing the TFNode with NGNode. Every NodeDef is different so might require more work to create a generic utility we can take it up as a separate task

Reply:
We will address the setting of unique name in another PR